lib/components/transaction_item.dart

```diff
@@ -0,0 +1,280 @@
+import 'package:finance_tracker/model/tag.dart';
+import 'package:intl/intl.dart';
+import 'package:flutter/services.dart';
+
+class TransactionItem extends StatefulWidget {
+  final Transaction transaction;
+  final List<Tag> tagList;
+  final List<MultiSelectItem<Tag>> items;
+  final FileController fileController;
+
+  const TransactionItem({
+    super.key,
+    required this.transaction,
+    required this.tagList,
+    required this.items,
+    required this.fileController,
+  });
+
+  @override
+  TransactionItemState createState() => TransactionItemState();
+}
+
+class TransactionItemState extends State<TransactionItem> {
+  late ValueNotifier<String> transactionName;
+  late ValueNotifier<String> transactionDate;
+  late ValueNotifier<List<Tag>> selectedTags;
+  late ValueNotifier<double> transactionAmount;
+  bool isExpanded = false;
+
+  @override
+  void initState() {
+    super.initState();
+    transactionName = ValueNotifier(widget.transaction.transactionName);
+    transactionDate = ValueNotifier(widget.transaction.transactionDate);
+    selectedTags = ValueNotifier(widget.transaction.transactionTag.map((index) => widget.tagList[index]).toList());
+    transactionAmount = ValueNotifier(widget.transaction.transactionAmount);
+  }
+
+  void _resetValues() {
+    transactionName.value = widget.transaction.transactionName;
+    transactionDate.value = widget.transaction.transactionDate;
+    selectedTags.value = widget.transaction.transactionTag.map((index) => widget.tagList[index]).toList();
+    transactionAmount.value = widget.transaction.transactionAmount;
+  }
+
+  Future<void> _selectDate(BuildContext context) async {
+    DateTime? pickedDate = await showDatePicker(
+      context: context,
+      initialDate: DateTime.parse(transactionDate.value),
+      firstDate: DateTime(2000),
+      lastDate: DateTime(2101),
+      builder: (BuildContext context, Widget? child) {
+        return Theme(
+          data: ThemeData.dark().copyWith(
+            primaryColor: NexusColor.text,
+            colorScheme: const ColorScheme.dark(primary: NexusColor.text, onPrimary: NexusColor.background),
+          ),
+          child: child!,
+        );
+      },
+    );
+    if (pickedDate != null) {
+      transactionDate.value = DateFormat('yyyy-MM-dd').format(pickedDate);
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      decoration: const BoxDecoration(
+        color: NexusColor.background,
+        border: Border(
+          bottom: BorderSide(
+            color: NexusColor.inputs,
+            style: BorderStyle.solid,
+            strokeAlign: BorderSide.strokeAlignInside,
+          ),
+        ),
+      ),
+      child: ExpansionTile(
+        title: Text(
+          widget.transaction.transactionName,
+          style: const TextStyle(color: NexusColor.text, fontSize: 20),
+        ),
+        subtitle: Text(
+          widget.transaction.transactionDate,
```

Comment:
wäre es nicht gut das Datum im init state direkt zu convertieren in das jeweilige Datumformat?
quasi sowas wie:
transactionDate = DateFormat('yyyy-MM-dd').format(widget.transaction.transactionDate);
  date.text = transactionDate!;

Reply:
Der Init state ist ja die existierende Transaction, wo das Format bereits korrekt ist. daher muss man das nicht in ein Format schreiben. Alle eingegebenen werte müssen dann halt darauf angepasst werden

Reply:
ja gut aber dachte nur das es vielleicht gut wäre die sichtbaren Daten irgendwie ersichtlicher darzustellen